zebra/zebra_routemap.c

```diff
@@ -1696,43 +1751,65 @@ static void zebra_route_map_event(route_map_event_t event,
 }
 
 /* ip protocol configuration write function */
-void zebra_routemap_config_write_protocol(struct vty *vty)
+void zebra_routemap_config_write_protocol(struct vty *vty,
+					  struct zebra_vrf *zvrf)
 {
 	int i;
 
 	for (i = 0; i < ZEBRA_ROUTE_MAX; i++) {
-		if (proto_rm[AFI_IP][i])
-			vty_out(vty, "ip protocol %s route-map %s\n",
-				zebra_route_string(i), proto_rm[AFI_IP][i]);
-
-		if (proto_rm[AFI_IP6][i])
-			vty_out(vty, "ipv6 protocol %s route-map %s\n",
-				zebra_route_string(i), proto_rm[AFI_IP6][i]);
-
-		if (nht_rm[AFI_IP][i])
-			vty_out(vty, "ip nht %s route-map %s\n",
-				zebra_route_string(i), nht_rm[AFI_IP][i]);
-
-		if (nht_rm[AFI_IP6][i])
-			vty_out(vty, "ipv6 nht %s route-map %s\n",
-				zebra_route_string(i), nht_rm[AFI_IP6][i]);
+		if (PROTO_RM_NAME(zvrf, AFI_IP, i))
+			vty_out(vty, (zvrf_id(zvrf) == VRF_DEFAULT)
```

Comment:
In other parts of the code I create a `char space[3]` and set it appropriately before the for loop and then I do this:
vty_out(vty, "%sip protocol %s route-map %s\n", space, zebra_route_string(i)...)  I think it makes it a bit cleaner